src/application.ts

```diff
@@ -111,10 +106,15 @@ export default class Application {
     }
 
     if (Array.isArray(data)) {
-      return Promise.all(data.map(record => this.serializeResources(record)));
+      const resource = await this.resourceFor(data[0].type);
```

Comment:
@spersico This might not work if you have different included types. Suppose you request a list books, including `authors` and `sellers`. If I'm reading this correctly, you're gonna be assuming every resource is an `author`.

```ts
// Minor enhancement to always work with arrays.
const arraify = (val) => Array.isArray(val) ? val : [val]:

return Promise.all(
  arraify(data).map(async record => 
    this.serializeResource(record, await this.resourceFor(record.type))
  )
);
```

Reply:
> @spersico This might not work if you have different included types. Suppose you request a list books, including authors and sellers. If I'm reading this correctly, you're gonna be assuming every resource is an author.

This function (serializeResources(data)) affects only to the main resource of the operation, meaning, that for your example this would only affect books. 
What it IS true, tough, is that I assume that an operation has only one main resource type, so the data parameter would only have ONE type of resource (the included resources are handled separately).

Reply:
I'm wrong in assuming that?

Reply:
I pushed the arraify approach, I liked that :-)

Reply:
Nope, your assumption is correct :) Glad you liked the arraify thingy